### PR TITLE
Android: add support for TextInputType.none

### PR DESCRIPTION
**Pull request by jpnurmi**
`TextInputType.none` makes it possible to disable the virtual keyboard for certain TextFields, and lays the foundations for custom in-app virtual keyboards (flutter/flutter#76072).

Ref: flutter/flutter#83567

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by jpnurmi** [comment]
https://user-images.githubusercontent.com/140617/120760346-4dc97c00-c514-11eb-9c26-05172d07477a.mp4

**Comment by jpnurmi** [comment]
Is there any reasonable way to test this?

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 1]
```diff
@@ -363,8 +365,12 @@ public void sendTextInputAppPrivateCommand(String action, Bundle data) {
   }
 
   private void showTextInput(View view) {
-    view.requestFocus();
-    mImm.showSoftInput(view, 0);
+    if (configuration.inputType.type != TextInputChannel.TextInputType.NONE) {
```
Comment (LongCatIsLooong) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java:368]: I would recommend not creating an input connection at all if you don't need the keyboard. See: https://developer.android.com/reference/android/view/View#onCreateInputConnection(android.view.inputmethod.EditorInfo)

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 2]
```diff
@@ -382,7 +395,11 @@ private void hideTextInput(View view) {
   void setTextInputClient(int client, TextInputChannel.Configuration configuration) {
     // Call notifyViewExited on the previous field.
     notifyViewExited();
-    inputTarget = new InputTarget(InputTarget.Type.FRAMEWORK_CLIENT, client);
+    if (canShowTextInput()) {
```
Comment (LongCatIsLooong) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java:398]: `canShowTextInput` depends on `this.configuration` which hasn't been updated at this point.

**Review by LongCatIsLooong — COMMENTED** [review]
Could you add tests to verify this works? E.g., when the type is "none" the flutter view returns null for `createInputConnection`.

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java (hunk 2 above)
Comment (jpnurmi) [reply shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java:399]: The result of blindly fixing failing tests :facepalm:

**Comment by chinmaygarde** [comment]
This needs to be rebased on ToT so that presubmits will pass after an infra failure. @LongCatIsLooong you comments looks like they have been addressed. Another review pass please?

**Review by LongCatIsLooong — APPROVED** [review]

**Pull request merged**